### PR TITLE
removed dependencies from browser object in before hooks where possible

### PR DESCRIPTION
**Pull request by vlad9719**
**What has been done**

Adjusted the `beforeSession` hook to check for the presence of a `browser` object

**Why**

To facilitate possibility of running test cases in the _watch_ mode. Without that check, the running of a test would fail because the `browser` global object would be undefined in the `beforeSession` hook, causing the code inside the hook to fail, when a test case is restarted in the _watch_ mode

### CODE REVIEW

**Review by heraldantony — APPROVED** [review]
looks good

File: src/scripts/dataExchange/dataExchange.ts [hunk 1]
```diff
@@ -7,64 +7,37 @@
...
-        // @ts-ignore
-        await importExportDataUtil.readData(fileOrDir, [param]);
-      } else {
-        delete importParams[param];
-        console.warn(
-          `"${fileOrDir}" does not exist or is not readable. Please check path or permissions.`
-        );
-      }
+    const importParams = config.params.import || {};
+    if (importParams) {
```
Comment (DIFSRIP) [inline src/scripts/dataExchange/dataExchange.ts:29]: this can never be falsy because of {}

File: src/scripts/dataExchange/dataExchange.ts [hunk 2]
```diff
@@ -212,3 +185,51 @@ class DataExchange {
...
+        delete importParams[param];
+        console.warn(
+          `"${fileOrDir}" does not exist or is not readable. Please check path or permissions.`
+        );
+      }
+    } 
+}
+
+async function readExportParams(config: Record<string, any>, exportParams: any) {
+  if (!exportParams) {
```
Comment (DIFSRIP) [inline src/scripts/dataExchange/dataExchange.ts:212]: check is redundant: https://github.com/SAP/wdio-qmate-service/pull/277/files?diff=split&w=0#diff-bb5bafb01127841195d1c3c05a6ff59f0eb078f7ec1cb36892c2f179b6a5491dR38

**Review by DIFSRIP — CHANGES_REQUESTED** [review]
Please see my comments

File: src/scripts/dataExchange/dataExchange.ts (hunk 2 above)
Comment (vlad9719) [reply src/scripts/dataExchange/dataExchange.ts]: removed the check

**Review by DIFSRIP — APPROVED** [review]

**Pull request merged**